client/state/sites/test/selectors.js

```diff
@@ -1184,4 +1228,778 @@ describe( 'selectors', () => {
+					options: {
+						jetpack_version: smallerVersion
+					}
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if is a multi-network site', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: true,
+					is_multi_network: true,
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if is not a main network site (urls don\'t match)', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: true,
+					URL: 'https://jetpacksite.me',
+					is_multisite: true,
+					options: {
+						is_multi_network: false,
+						unmapped_url: 'https://example.wordpress.com',
+						main_network_site: 'https://anotherexample.wordpress.com'
+					}
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if `disallow_file_mods` is disabled', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: true,
+					URL: 'https://jetpacksite.me',
+					is_multisite: true,
+					options: {
+						is_multi_network: false,
+						unmapped_url: 'https://example.wordpress.com',
+						main_network_site: 'https://example.wordpress.com',
+						file_mod_disabled: [
+							'disallow_file_mods',
+						]
+					}
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `false` if `has_no_file_system_write_access` is disabled', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: true,
+					URL: 'https://jetpacksite.me',
+					is_multisite: true,
+					options: {
+						is_multi_network: false,
+						unmapped_url: 'https://example.wordpress.com',
+						main_network_site: 'https://example.wordpress.com',
+						file_mod_disabled: [
+							'has_no_file_system_write_access',
+						]
+					}
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
+		} );
+
+		it( 'it should return `true` for the site right configurations', () => {
+			const state = createStateWithItems( {
+				[ siteId ]: {
+					ID: siteId,
+					jetpack: true,
+					URL: 'https://jetpacksite.me',
+					is_multisite: true,
+					options: {
+						is_multi_network: false,
+						unmapped_url: 'https://example.wordpress.com',
+						main_network_site: 'https://example.wordpress.com',
+						file_mod_disabled: []
+					}
+				}
+			} );
+
+			const canUpdateFiles = canJetpackSiteUpdateFiles( state, siteId );
+			expect( canUpdateFiles ).to.equal( false );
```

Comment:
Code check asserts `false`. Test description says `true`. Probably it needs some tweaks.